sensor/common/clusterentities/store_test.go

```diff
@@ -284,3 +368,321 @@ func (s *ClusterEntitiesStoreTestSuite) TestChangingIPsAndExternalEntities() {
+				0: {
+					{
+						deploymentID: "depl1",
+						containerID:  "pod1",
+						incremental:  true,
+					},
+				},
+				5: {
+					{
+						deploymentID: "depl1",
+						containerID:  "pod1",
+						incremental:  false, // overwrite
+					},
+				},
+			},
+			operationAfterTick: map[int]operation{1: mapReset},
+			containerIDsAfterTick: []map[string]whereContainerIDisStored{
+				{"pod1": theMap}, // before tick 1: container should be added immediately
+				{"pod1": theMap}, // after tick 1
+				// reset
+				{"pod1": history}, // after tick 2: will remember that for one more tick
+				{"pod1": history}, // after tick 3: will remember that for this last tick
+				{"pod1": nowhere}, // after tick 4: history expired - should be forgotten
+				{"pod1": theMap},  // after tick 5: re-added pod1 should be findable from now on until the next reset
+				{"pod1": theMap},  // after tick 6: no further reset was planned, so we should find pod1 forever
+				{"pod1": theMap},
+				{"pod1": theMap},
+				{"pod1": theMap},
+			},
+		},
+		"Container is deleted normally": {
+			numTicksToRemember: 2,
+			entityUpdates: map[int][]eUpdate{
+				0: {
+					{
+						deploymentID: "depl1",
+						containerID:  "pod1",
+						incremental:  true,
+					},
+				},
+			},
+			operationAfterTick: map[int]operation{1: deleteDeployment1},
+			containerIDsAfterTick: []map[string]whereContainerIDisStored{
+				{"pod1": theMap}, // before tick 1: container should be added immediately
+				{"pod1": theMap}, // after tick 1
+				// container deletion
+				{"pod1": history}, // after tick 2: will remember that for one more tick
+				{"pod1": history}, // after tick 3: will remember that for this last tick
+				{"pod1": nowhere}, // after tick 4: history expired - should be forgotten forever
+				{"pod1": nowhere},
+			},
+		},
+		"Re-adding normally deleted container should reset the history status": {
+			numTicksToRemember: 2,
+			entityUpdates: map[int][]eUpdate{
+				0: {
+					{
+						deploymentID: "depl1",
+						containerID:  "pod1",
+						incremental:  true,
+					},
+				},
+				5: {
+					{
+						deploymentID: "depl1",
+						containerID:  "pod1",
+						incremental:  true,
+					},
+				},
+			},
+			operationAfterTick: map[int]operation{1: deleteDeployment1},
+			containerIDsAfterTick: []map[string]whereContainerIDisStored{
+				{"pod1": theMap}, // before tick 1: container should be added immediately
+				{"pod1": theMap}, // after tick 1
+				// container deletion
+				{"pod1": history}, // after tick 2: will remember that for one more tick
+				{"pod1": history}, // after tick 3: will remember that for this last tick
+				{"pod1": nowhere}, // after tick 4: history expired - should be forgotten forever
+				// adding container again
+				{"pod1": theMap}, // after tick 5: should be normally added to the map
+				{"pod1": theMap}, // after tick 6: should stay in the map until the next deletion or reset
+				{"pod1": theMap},
+			},
+		},
+	}
```

Comment:
It would be interesting to add a case where the container gets re-added before the history expires:

```golang
{"pod1": theMap}, // before tick 1: container should be added immediately
{"pod1": theMap}, // after tick 1
// container deletion
{"pod1": history}, // after tick 2: will remember that for one more tick
// adding container again
{"pod1": theMap}, // after tick 3: should be normally added to the map
```

Reply:
Added in [3833688](https://github.com/stackrox/stackrox/pull/13155/commits/3833688c5daa08d2c2693863b85fdd4440b3998c)